### DISCUSSION TITLE
Why return 2 * fx when cheirality exception is met in GenericProjectionFactor?

### DISCUSSION THREAD

**Discussion by amaysaxena** [borglab/gtsam#1191] [Q&A]
Hello,

In `GenericProjectionFactor`, we have the following handling for when `camera.project` throws a `CheiralityException`:
```C++
} catch(CheiralityException) {
         if (H1) *H1 = Matrix::Zero(2,6);
         if (H2) *H2 = Matrix::Zero(2,3);
         if (verboseCheirality_)
           // Stuff
         if (throwCheirality_)
            // Other stuff
}
return Vector2::Constant(2.0 * K_->fx());
```
What is the rationale behind returning `(2 fx, 2 fx)` as the error vector? Why is this a better choice than anything else? Would appreciate if someone could help me understand this a bit better.

**Comment by ProfFan** [comment]
I actually have the same question. This fragment of code is really old (11 years ago!) by @richardroberts . Origin is at https://github.com/borglab/gtsam/commit/3d68672b20d168e57b51401802c3bcb334dfe578 . Unfortunately no explanation is given in the commit comments.

Pinging @dellaert @varunagrawal for help.
